aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -115,9 +115,9 @@ static void cumprod_cpu_kernel(const Tensor& result, const Tensor& self, int64_t
 // custom min and max to be used in logcumsumexp for complex arguments
 template <typename scalar_t, bool min>
 c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y) {
-  if (std::isnan(y)) {  // either real is nan or imag is nan
+  if (at::_isnan(y)) {  // either real is nan or imag is nan
```

Comment:
These are CPU kernels, is _isnan available here?

Reply:
I think we can use `_isnan` on both cpu & gpu sides. Ref: #18021.